ChurchCRM/CRM
# 1 feature request migrate hardcoded states dropdown

# Description & Issue number it closes 
This should be considered a *DRAFT* Pull Request for Issue #6644.  It was chosen because of the 'good first issue' tag, and I was looking for a relatively simple fix that would help me navigate the waters for future contributions.  The main purpose of this fix is to consolidate where State DropDown options are presented, and use them in a consistent way.

## Screenshots (if appropriate)
I don't think this (as it currently stands...) should affect anything related to screenshots.  Though there could be later changes after discussions that might make some small changes.

## How to test the changes?
What I've been doing to test this is just going into the view of "Active People" and selecting someone to edit, which goes through the PersonEditor.php.  With the 'old' hard-coded way from the original StateDropDown.php, the list of States provided (for US) started with Alabama, Alaska, Arizona.....  After the change, you should see: Alabama, Alaska, American Samoa, ... as this comes from the list in `src/locale/states/us.json`

## Type of change


# How Has This Been Tested?
See above on "How to test the changes".  Note that I have only tested this running in a Dev environment, using docker -- but I don't think that matters for this type of change.

# Checklist:

   * Possibly a new log message is produced now.
   * I'm unclear how to proceed on this, but open to suggestions
   * Similarly, unclear on how to run these tests currently.
- [N/A] Any dependent changes have been merged and published in downstream modules

I wanted to initiate this Pull Request to get some formal code review here.  This is my first time contributing on GitHub, so I'm not sure how this will go, but I would like to see review-comments on my existing changes, and I have a couple of followup questions I'd like to discuss before this is considered by me ready-to-merge.  I'm hoping that can be done in the form of comments in the GitHub issues somehow -- but as this is my first, waiting to see how that works out.
